HousingRepairsOnlineApi/Helpers/InternalEmailSender.cs

```diff
@@ -9,24 +9,20 @@ public class InternalEmailSender : IInternalEmailSender
     {
         private IRetrieveImageLinkUseCase retrieveImageLinkUseCase;
         private ISendInternalEmailUseCase sendInternalEmailUseCase;
+        private INotificationConfigurationResolver notificationConfigurationResolver;
 
-        public InternalEmailSender(IRetrieveImageLinkUseCase retrieveImageLinkUseCase, ISendInternalEmailUseCase sendInternalEmailUseCase)
+        public InternalEmailSender(IRetrieveImageLinkUseCase retrieveImageLinkUseCase, ISendInternalEmailUseCase sendInternalEmailUseCase, INotificationConfigurationResolver notificationConfigurationResolver)
         {
             this.retrieveImageLinkUseCase = retrieveImageLinkUseCase;
             this.sendInternalEmailUseCase = sendInternalEmailUseCase;
+            this.notificationConfigurationResolver = notificationConfigurationResolver;
         }
 
         public async Task Execute(Repair repair)
         {
-            var imageLink = "None";
-            if (!String.IsNullOrEmpty(repair.Description?.PhotoUrl))
-            {
-                await Task.Run(() =>
-                {
-                    imageLink = retrieveImageLinkUseCase.Execute(repair.Description?.PhotoUrl);
-                });
-            }
-            sendInternalEmailUseCase.Execute(repair.Id, repair.Address.LocationId, repair.Address.Display, repair.SOR, repair.Description.Text, repair.ContactDetails?.Value, imageLink);
+            var sendNotification = notificationConfigurationResolver.Resolve(repair.RepairType);
+            var imageLink = await sendNotification.GetImageLink(retrieveImageLinkUseCase, repair);
```

Comment:
Was it not possible to utilise `GetImageLink` within `INotificationConfigurationProvider.GetPersonalisationForInternalEmailTemplate` rather than call it here?

Reply:
I have now moved this into INotificationConfigurationProvider